### PR TITLE
[CDAP-14552] Modifies textarea widget to have line numbers

### PR DESCRIPTION
**Pull request by tonybach**
JIRA: https://issues.cask.co/browse/CDAP-14552
Builds: https://builds.cask.co/browse/CDAP-UDUT146

### CODE REVIEW

**Comment by elfenheart** [comment]
Don't you have to modify the gulpfile to include the plain_text mode?

**Comment by tonybach** [comment]
Hmm I didn't have to and it still worked when I tested it out, but have added it to be safe.

**Comment by bdmogal** [comment]
@tonybach need to update docs as well right?

**Comment by nitinmotgi** [comment]
Does it provides line number ?

**Comment by nitinmotgi** [comment]
Also why are we not improving text area

**Comment by tonybach** [comment]
@nitinmotgi Yes this provides line number. The library we use for Javascript/Python/Scala code editors also provide plain text mode, so it was much easier to use that than to add line numbers to textarea.

File: cdap-ui/app/directives/widget-container/widget-js-editor/widget-js-editor.js [hunk 1]
```diff
@@ -34,10 +43,22 @@ angular.module(PKG.name + '.commons')
...
           }
         };
-
+        let height;
+        // textarea widget
+        if ($scope.mode && $scope.mode === 'plain_text') {
+          height = DEFAULT_TEXTAREA_HEIGHT;
+          if ($scope.rows) {
+            let computedHeight = $scope.rows * LINE_HEIGHT;
+            height = Math.max(computedHeight, height);
```
Comment (elfenheart) [inline cdap-ui/app/directives/widget-container/widget-js-editor/widget-js-editor.js]: I don't think we should do a `Math.max` here. If the user already specified the number of rows, we should honor that regardless, except perhaps negative number or `0` (which we should validate against).

**Review by elfenheart — DISMISSED** [review]
👍 LGTM

**Comment by tonybach** [comment]
Latest build is green: https://builds.cask.co/browse/CDAP-UDUT146-4

**Review by elfenheart — APPROVED** [review]

**Pull request merged**